### PR TITLE
Making migration generator generate namespaced migrations

### PR DESCRIPTION
**Pull request by bizley**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Tests pass?   | ✔️
| Fixed issues  | #17437 

This allows generator to understand camelcased commands.

1. Namespaced class name is generated according to namespaced format `M<YYMMDDHHMMSS><Name>` where `Name` is camelized version of what is provided for generator.
2. Table names provided for generator are underscored in migration (so `User` is `user`, `MyData` is `my_data`, and `BOB` is `b_o_b` unless the name is preceded with underscore (for BC) then name case is kept as-is.
3. If table name should be suffixed with underscore you must use two underscores for generator.
4. Unintentional change but I don't want to fight it - now you can run generator with any mix of cases and/or underscores separating command words like `create_userTable`, `CreateUsertabLE` and so on. In guide I'm using official format every time.
5. Guide is updated for this (also PL version is added).

### CODE REVIEW

**Comment by samdark** [comment]
Tests do not pass.

**Comment by bizley** [comment]
Fixed now ;)

File: framework/console/controllers/MigrateController.php [hunk 1]
```diff
@@ -375,6 +376,26 @@ protected function getMigrationNameLimit()
...
     }
 
+    /**
+     * Normalizes table name for generator.
+     * When name is preceded with underscore name case is kept - otherwise it's converted from camelcase to underscored.
+     * Last underscore is always trimmed so if there should be underscore at the end of name use two of them.
+     * @param string $name
+     * @return string
+     */
+    private function normalizedTableName($name)
```
Comment (samdark) [inline framework/console/controllers/MigrateController.php]:
```suggestion
    private function normalizeTableName($name)
```

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
All good except minor method naming issue.

File: framework/console/controllers/MigrateController.php (hunk 1 above)
Comment (bizley) [reply framework/console/controllers/MigrateController.php]: Right, thanks! Commited as suggested

**Comment by samdark** [comment]
Merged. Thank you!

**Pull request merged**